core/domain/collection_services_test.py

```diff
@@ -116,6 +117,180 @@ def test_get_collection_titles_and_categories(self):
                 })
 
 
+class ExplorationInCollectionContextUnitTests(CollectionServicesUnitTests):
+    """Tests query and recording methods related to explorations which are
+    played in the context of a collection.
+    """
+
+    COL_ID_0 = '0_collection_id'
+    COL_ID_1 = '1_collection_id'
+    EXP_ID_0 = '0_exploration_id'
+    EXP_ID_1 = '1_exploration_id'
+    EXP_ID_2 = '2_exploration_id'
+
+    def _get_completion_model(self, user_id, collection_id):
+        return user_models.ExplorationInCollectionCompletionModel.get(
+            user_id, collection_id)
+
+    def _record_completion(self, user_id, collection_id, exploration_id):
+        collection_services.record_played_exploration_in_collection_context(
+            user_id, collection_id, exploration_id)
+
+    def setUp(self):
+        super(ExplorationInCollectionContextUnitTests, self).setUp()
+
+        # Create a new collection and exploration.
+        self.save_new_valid_collection(
+            self.COL_ID_0, self.OWNER_ID, exploration_id=self.EXP_ID_0)
+
+        # Create another two explorations and add them to the collection.
+        for exp_id in [self.EXP_ID_1, self.EXP_ID_2]:
+            self.save_new_valid_exploration(exp_id, self.OWNER_ID)
+            collection_services.update_collection(
+                self.OWNER_ID, self.COL_ID_0, [{
+                    'cmd': collection_domain.CMD_ADD_COLLECTION_NODE,
+                    'exploration_id': exp_id
+                }], 'Added new exploration')
+
+    def test_get_completed_exploration_ids(self):
+        # There should be no exception if the user or collection do not exist.
```

Comment:
Good docstring suggest. Updated my previous attempt. Updated the docstring in storage/user/gae_models.py for ExplorationInCollectionCompletionModel and added  aTODO to fix it later.

Also, from offline discussions and referring to this model elsewhere in the codebase, I've decided to rename it to 'CollectionProgressModel' since it seems to be more fitting for what's happening. This also creates a pattern that could extend to explorations and allows us to extend it with more information in the future.